### ISSUE TITLE
query for lights returns "unknown state" for dimmer

### ISSUE THREAD

**Issue by mdopp** [openhab/openhab-google-assistant#23]
I have 12 items that are marked as "Lighting". When I ask google "Which Lights are currently on" it returns that "The light Terrassenlicht is turned off. 11 Lights are currently not reachable".

Reason: "Terrassenlich" is a Switch, which returns "ON" or "OFF" as a state. All others are Dimmers, returning an value between "0" - "100".

I guess that the QUERY should not only return "on" with true or false, but also brightness and all other parameters that are set by the traits

```
"on": true,
"brightness": 80,
"color": { "name": "cerulean", "spectrumRGB": 31655 }
```

So to do this correclty, the traits must be known and the needed values should be returned here.

**Comment by skimdankish** [comment]
I think it depends on the request, when I ask if a Dimmer item is on or off, it is using the OnOff trait..which only requires very simple information to be returned see attached.

https://developers.google.com/actions/smarthome/traits/onoff#sample-query-request-and-response

**Comment by mdopp** [comment]
@skimdankish you are right that it should be kept simple. But we inform google that we also support "Brightness" for "Dimmer", hence we should also answer a query for this in the correct way.
About your suggestion: This works, but does not respect the traits and the device type. Solving it this way might break functionality in the future. Still it is better than my solution.

**Comment by skimdankish** [comment]
Take a look here, I have been running off this:
https://github.com/skimdankish/openhab-google-assistant/blob/master/functions/openhab.js

Look at line 131, it handles Dimmer requests. Not sure if it's the right way, but this works...

**Comment by mdopp** [comment]
That looks good enough. But you make new assumptions which values should be delivered, while we did this evaluation already when we told Google about the supported traits. But still: far better than the actual version

**Comment by skimdankish** [comment]
Can you please let me know where? I have been testing this in my own bubble for a few weeks with nobody to review - perhaps I can tweak.  I felt that there was a disconnect between OH item type and tags vs the Google device type/trait.  So I attempted to compensate, perhaps there's a better way...I just had a very cold month so it was nice to have the Thermostat function working :)

**Comment by mdopp** [comment]
i have now tried to include your pull request #20 and #18 in my #24. This how I meant it. But I can not test it. As yours is working, would you like to integrate my changes into yours? Then I will delete my pull request.

**Comment by skimdankish** [comment]
Got it running, had to make a few edits, I can comment on the pull request if you don't mind.
1) Need to definte itemData before the switch statement and checking the traits
2) Thermostat/CurrentTemperature just had the traits assigned in SyncAndDiscover function, so it needed to be added manually for this into getSwitchableTraits function
3) The utils.js from my fork has the correct thermostat modes formatting.  Google wants lower case, there seems to be some leftover code from the Alexa skill which is where this is from.

**Comment by skimdankish** [comment]
The adjustThermostat functions from your build are from the original.  I had to edit them quite a bit to work.  I updated my fork https://github.com/skimdankish/openhab-google-assistant/blob/master/functions 
- openhab.js (some housekeeping, like I mentioned above)
- I just kept my edited Thermostat functions (to get thermostat info and to properly update it)

I have confirmed this works just as well as my build was working with the new backend code with respect to traits.
